cht/adapter.py

```diff
@@ -2,307 +2,292 @@
-        self.preciceFaceCentersMeshID = self.precice.getMeshID(self.faceCentersMeshName)
-        self.precice.setMeshVertices(self.preciceFaceCentersMeshID, len(self.faceCenterCoordinates), np.hstack(self.faceCenterCoordinates), self.preciceFaceCenterIndices)
-        
-    def setDataIDs(self, names):
-        for writeDataName in names["write-data"]:
-            if writeDataName.find("Heat-Transfer-Coefficient-") >= 0:
-                self.writeHCoeffDataID = self.precice.getDataID(writeDataName, self.preciceNodesMeshID)
-            elif writeDataName.find("Sink-Temperature-") >= 0:
-                self.writeTempDataID = self.precice.getDataID(writeDataName, self.preciceNodesMeshID)
-        for readDataName in names["read-data"]:
-            if readDataName.find("Heat-Transfer-Coefficient-") >= 0:
-                self.readHCoeffDataID = self.precice.getDataID(readDataName, self.preciceFaceCentersMeshID)
-            elif readDataName.find("Sink-Temperature-") >= 0:
-                self.readTempDataID = self.precice.getDataID(readDataName, self.preciceFaceCentersMeshID)
-
-    def getPreciceNodeIndices(self):
-        return self.preciceNodeIndices
-    
-    def getPreciceFaceCenterIndices(self):
-        return self.preciceFaceCenterIndices
-    
-    def getPreciceNodesMeshID(self):
-        return self.preciceNodesMeshID
-    
-    def getPreciceFaceCentersMeshID(self):
-        return self.preciceFaceCentersMeshID
-    
-    def getNodeCoordinates(self):
-        return self.nodeCoordinates
-    
-    def getFaceCenterCoordinates(self):
-        return self.faceCenterCoordinates
-        
-    def getNormals(self):
-        return self.normals
-    
-    def createBCs(self):
-        """
+	def __init__(self, precice, participantName, names, MESH, SHMESH, MODEL, MAT, isNonLinear = False):
+
+		self.precice = precice
+		self.participantName = participantName
+
+		self.groupName = ""
+		self.facesMeshName = ""
+		self.nodesMeshName = ""
+
+		self.nodes = []
+		self.faces = []
+		self.connectivity = []
+		self.nodeCoordinates = []
+		self.faceCenterCoordinates = []
+		self.normals = None
+
+		self.isNonLinear = isNonLinear
+		self.conductivity = None
+		self.isConductivityInitialized = False
+		self.delta = 1e-5
+
+		self.preciceNodeIndices = []
+		self.preciceFaceCenterIndices = []
+
+		self.preciceFaceCentersMeshID = 0
+		self.preciceNodesMeshID = 0
+
+		self.readHCoeffDataID = 0
+		self.readTempDataID = 0
+		self.writeHCoeffDataID = 0
+		self.writeTempDataID = 0
+
+		self.readTemp = []
+		self.readHCoeff = []
+		self.writeTemp = []
+		self.writeHCoeff = []
+
+		self.readDataSize = 0
+		self.writeDataSize = 0
+
+		self.MESH = MESH
+		# Shifted mesh (contains only the interface, and is shifted by delta in the direction opposite to the normal)
+		self.SHMESH = SHMESH
+		self.MODEL = MODEL
+		self.MAT = MAT
+		self.LOAD = None
+		self.LOADS = []
+		self.mesh = partition.MAIL_PY()
```

Comment:
Why was this needed?

Reply:
Because the imports were changed (see line 7). Utilai is now imported as a whole. To get the contents of MAIL_PY(), we need to call it by partition.MAIL_PY()